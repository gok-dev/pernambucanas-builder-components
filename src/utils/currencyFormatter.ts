export const currencyFormatter = (amount: number) => {
  return parseFloat(amount.toString()).toLocaleString('pt-BR', {
    minimumFractionDigits: 2,
    style: 'currency',
    currency: 'BRL'
  })
}
