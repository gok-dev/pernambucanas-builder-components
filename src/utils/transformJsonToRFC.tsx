import { Banner } from 'index'
import * as React from 'react'

const rFCStringMapper = {
  Banner: Banner
}

export const transformJsonToRFC = (
  json: { name: string; props: any }[],
  onChange: (param: any) => void
): React.ReactElement[] => {
  const Components = json?.map(item => {
    return React.createElement(rFCStringMapper[item.name], {
      ...item.props,
      onChange
    })
  })

  return Components
}
