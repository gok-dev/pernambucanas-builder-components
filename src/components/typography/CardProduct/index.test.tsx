import React from 'react'
import { renderWithTheme } from 'utils/renderWithTheme'
import { CardProduct } from './index'

describe( '<Card />', () => {
  it( 'shold render', () => {
    const { container } = renderWithTheme(
      <CardProduct
        title="Produto"
        amount={999}
        image={
          <img
            src="/assets/product.png"
            alt="Ventilador Britânia BVT410 Turbo Preto"
            title="Ventilador Britânia BVT410 Turbo Preto"
            width="220"
          />
        }
      />
    )

    expect( container.firstChild ).toMatchSnapshot()
  } )
} )
