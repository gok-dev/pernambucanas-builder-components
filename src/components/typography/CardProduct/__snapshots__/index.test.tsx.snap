// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> shold render 1`] = `
.c0 {
  background-color: #FFFFFF;
  border-radius: 8px;
  width: 100%;
  min-height: 100px;
  padding: 8px;
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0:hover {
  cursor: pointer;
  box-shadow: 2px 2px 10px rgba(0,0,0,0.17);
}

.c5 {
  font-size: 34px;
  font-size: 18px;
  font-weight: 500;
}

.c6 {
  font-weight: 400;
  margin-top: 8px;
  margin-bottom: px;
  margin-right: px;
  margin-left: px;
  font-size: 12px;
}

.c6 a {
  color: #2485EA;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c6 a:hover {
  color: #1C67B8;
}

.c6 mark {
  border-radius: 3px;
  padding: 3px 4px;
  font-size: 12px;
}

.c6 strong {
  font-size: 12px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  position: relative;
}

.c2 button {
  position: absolute;
  right: 0;
  top: 0;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 img {
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 6px;
  margin-bottom: 8px;
}

.c3 {
  padding: 8px;
}

@media (min-width:460px) {
  .c4 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c4 p {
    margin-left: 6px;
  }
}

<article
  class="c0"
>
  <div
    class="c1"
  >
    <figure
      class="c2"
    >
      <img
        alt="Ventilador Britânia BVT410 Turbo Preto"
        src="/assets/product.png"
        title="Ventilador Britânia BVT410 Turbo Preto"
        width="220"
      />
    </figure>
    <div
      class="c3"
    >
      <header>
        <div
          class="c4"
        >
          <h1
            class="c5"
            size="18"
          >
            R$ 999,00
          </h1>
        </div>
      </header>
      <p
        class="c6"
        size="12"
      >
        Produto
      </p>
    </div>
  </div>
</article>
`;
